pkg/shell/nav.tsx

```diff
@@ -60,60 +117,59 @@ export class CockpitNav extends React.Component {
         };
 
         this.clearSearch = this.clearSearch.bind(this);
+        this.props = props;
     }
 
     componentDidMount() {
-        const self = this;
         const sel = this.props.selector;
         // Click on active menu item (when using arrows to navigate through menu)
         function clickActiveItem() {
             const cur = document.activeElement;
-            if (cur.nodeName === "INPUT") {
-                const el = document.querySelector("#" + sel + " li:first-of-type a");
+            if (cur instanceof HTMLInputElement) {
+                const el = document.querySelector<HTMLElement>("#" + sel + " li:first-of-type a");
                 if (el)
                     el.click();
-            } else {
+            } else if (cur instanceof HTMLElement) {
                 cur.click();
             }
```

Comment:
For our own sanity, this could have an `else console.error()` so that tests fail if we forget something here.

Reply:
Done.